projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx

```diff
@@ -0,0 +1,91 @@
+import { Component, Element, Event, EventEmitter, Listen, Prop } from '@stencil/core';
+import { DotCMSContentTypeField } from '../../../../dotcms/src/lib/models';
+import { DotFormFields } from './dot-form-fields';
+
+const fieldMap = {
+    Text : ((fieldParam: DotCMSContentTypeField) => DotFormFields.Text(fieldParam)),
+    Checkbox : ((fieldParam: DotCMSContentTypeField) => DotFormFields.Checkbox(fieldParam)),
+    Select : ((fieldParam: DotCMSContentTypeField) => DotFormFields.Select(fieldParam))
+};
+
+@Component({
+    tag: 'dot-form'
+})
+export class DotFormComponent {
+    @Element() el: HTMLElement;
+    @Event() formSubmit: EventEmitter;
+    @Prop() fields = [];
+    @Prop() fieldsToShow = [];
+    @Prop() resetLabel = 'Reset';
+    @Prop() submitLabel = 'Submit';
+
+    @Prop({ mutable: true }) _formValues = {};
```

Comment:
why just not "value", is a prop, is public, we don't need the `_` or the prefix.